src/RefocusCollectorEval.js

```diff
@@ -90,28 +89,49 @@ class RefocusCollectorEval {
 
     debug('evalUtils.safeToUrl returning: ${retval}');
     return retval;
-  }
+  } // safeToUrl
 
   /*
-   * Checks for a status code regex match which maps to a transform for
-   * error samples and returns the last error transform matched. If 200 is
-   * matched, it will override the default transform.
-   * @param {Object} tr - transform object.
+   * Returns the transform function for the given status, based on this order
+   * of precedence:
+   * (1) the first status code match from transform.errorHandlers
+   * (2) transform.default (for status 2xx only)
+   * (3) false
+   *
+   * @param {Object} transform - the sample generator template's transform object.
    * @param {String} status - Status code.
+   * @returns {Any} a transform function or false
    */
-  static statusCodeMatch(tr, status) {
-    let func;
-    if (tr.errorHandlers) {
-      Object.keys(tr.errorHandlers).forEach((statusMatcher) => {
-        const re = new RegExp(statusMatcher);
+  static getTransformFunction(transform, status) {
+    /*
+     * First check for a status match in transform.errorHandlers (including any
+     * "override" for any 2xx status the template may define).
+     */
+    if (transform.errorHandlers) {
+      /*
+       * Sort the errorHandlers keys alphabetically so there's a predictable
+       * order of evaluation, since we return the "first" match.
+       */
+      const statusMatchers = Object.keys(transform.errorHandlers).sort();
```

Comment:
I think we should maintain the order its defined in. So I can do
'404': () => {},
'4..': () => {},
and be sure that a 404 will be handled by the first one.

Reply:
Object.keys doesn't necessarily return the keys in the order in which they are defined:

console.log(Object.keys({ 404: '404', '4..': '4..' }))
// ["404", "4.."]

console.log(Object.keys({ '4..': '4..', '404': '404' }))
// ["404", "4.."]

console.log(Object.keys({ '4..': '4..', 500: '500', '404': '404' }))
// ["404", "500", "4.."]

console.log(Object.keys({ 500: '500', '4..': '4..', '404': '404' }))
// ["404", "500", "4.."]

Reply:
Good point. Ideally we would be able to support the case above though. Not sure if there's a good way to do that. Thoughts?

Maybe we could match all the \d\d\d ones first, then match the others if necessary?

Reply:
I like the idea of trying to do exact matches on \d\d\d first -- I'll do that.